### ISSUE TITLE
Cannot set reference for residual function in Python

### ISSUE THREAD

**Issue by edantec** [Simple-Robotics/aligator#209]
## Bug description

In Python, the SetReference function of residuals does not work in some cases, probably because the bindings return a copy of the object somewhere instead of a reference.

## Expected behavior

When holding a CostStack object in Python containing a residual, we should be able to modify the corresponding residual reference.

### Code

```
import numpy as np
import aligator
import pinocchio as pin
import example_robot_data as erd

robot = erd.load("talos")
rmodel = robot.model
RF_id = rmodel.getFrameId("right_sole_link")
nv = rmodel.nv
nu = nv - 6

space = aligator.manifolds.MultibodyPhaseSpace(rmodel)

posref = pin.SE3.Identity()
frame_fn_RF = aligator.FramePlacementResidual(space.ndx, nu, rmodel, posref, RF_id)
rcost = aligator.CostStack(space, nu)
residual = aligator.QuadraticResidualCost(space, frame_fn_RF, np.eye(6))
rcost.addCost(residual)

newpos = pin.SE3.Identity()
newpos.translation[2] = 0.2
print(
    "Translation stored = ",
    str(rcost.components[0][0].residual.getReference().translation),
)
rcost.components[0][0].residual.setReference(newpos)
print(
    "Set reference translation to ",
    str(newpos.translation),
)
print(
    "New translation stored = ",
    str(rcost.components[0][0].residual.getReference().translation),
)
```

## Additional context

Macro `ALIGATOR_EIGENPY_HAS_MAP_SUPPORT` is on.

### System

- OS: Ubuntu 20.04
- aligator version: 0.8.0

**Comment by edantec** [comment]
Addendum: this problem has been encountered before and was supposedly solved by this commit: 81dc2332a05cb71bad4d3bf49cb81a2fd25c62dd
The switch from vector to map of components probably caused the problem to arise again.

**Comment by ManifoldFR** [comment]
This is my fault, I forgot to overload the base_get_item in the map visitor... 🤦🏼‍♂️

**Comment by ManifoldFR** [comment]
Okay, we need to fix this like today

**Comment by ManifoldFR** [comment]
I just pushed on devel @edantec

**Comment by ManifoldFR** [comment]
Resolved by 36c4639c
